File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.HashMap;
class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int [targets.length];
        
        HashMap<Character,Integer> map = new HashMap<>();
        
        for(int i = 0 ; i < keymap.length ; i++){
            for(char ch : keymap[i].toCharArray()){
                int min = keymap[i].indexOf(ch)+1;
                if(!map.containsKey(ch) || map.get(ch)>min){
                    map.put(ch,min);                
                }
            }
        }
        
        for(int i = 0 ; i < targets.length ; i++){
            int num = 0;
            for(char ch : targets[i].toCharArray()){
                if(map.containsKey(ch)){
                    num +=map.get(ch);
                }else{
                    num = -1;
                    break;
                }
            }
            answer[i] +=num;
        }
        
        return answer;
    }
}

File: C#/프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.cs
using System;
using System.Collections.Generic;
using System.Linq;
public class Solution {
    public int[] solution(string[] keymap, string[] targets) {
 int[] answer = new int[targets.Length];

            for(int i = 0;i < targets.Length;i++)
            {
                foreach(char c in targets[i])
                {
                    List<int> keyLst = new List<int>();
                    for(int j = 0;j < keymap.Length; j++)
                    {
                        int index = keymap[j].IndexOf(c);
                        if (index != -1)
                        {
                            keyLst.Add(index + 1);
                        }
                    }
                    if (keyLst.Count == 0)
                    {
                        answer[i] = -1;
                        break;
                    }
                    else answer[i] += keyLst.Min();
                }
            }

            return answer;
    }
}